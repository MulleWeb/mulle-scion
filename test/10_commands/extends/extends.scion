{% extends "_base.scion" %}

This text is after extends, so it won't be printed.
It's a confusing way to comment...:)
All the blocks following are slurped up and remembered till the end of the 
enveloping template. Then blocks are substituted.

{% block header %}
this will be overridden shortly by the following block
{% endblock %}

{% block header %}
replaced the whole header
{% endblock %}

{% block body_extension %}
extended the body
{% endblock %}

{% block footer_extension %}
 {% block footer_extension2 %}
default footer_extension2 text
 {% endblock %}
{% endblock %}

{% block hackish_extension %}A bit of a mind bender{% endblock %}
 
{% extends "_hackish.scion" %}

