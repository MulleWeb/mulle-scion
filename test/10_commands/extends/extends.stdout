extends.scion: 1
_header.scion: 2
replaced the whole header
_header.scion: 7
_body.scion: 2
my body
extended the body
_body.scion: 7
_footer.scion: 2
my footer
it's tricky tricky tricky
_footer.scion: 7
_hackish.scion: 2
A bit of a mind bender but useful for doing a little multiple inheritance.
_hackish.scion: 8
_empty.scion: 2
